### PR TITLE
deps: Bump @sentry/javascript dependencies to 6.17

### PR DESCRIPTION
**Pull request by jennmueng**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps the @sentry/javascript dependencies to 6.17.9 and handles the internal breaking changes.
Closes https://github.com/getsentry/sentry-react-native/issues/2033

## :green_heart: How did you test it?
Local sample and see e2e tests. Updated unit tests, the span recorder now lists the transaction twice? @AbhiPrasad 

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I updated tests to verify changes
- [x] All tests passing
- [ ] No breaking changes (the `Status` enum is no longer exported and is instead `EventStatus`

### CODE REVIEW

File: src/js/wrapper.ts [hunk 1]
```diff
@@ -84,6 +83,9 @@ export const NATIVE: SentryNativeWrapper = {
 
     const event = this._processLevels(_event);
 
+    // Delete this metadata as this should not be sent to Sentry.
+    delete event.sdkProcessingMetadata;
```
Comment (marandaneto) [inline src/js/wrapper.ts:87]: is this a new thing?

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
@bruno-garcia this likely requires at least a minor bump, maybe even major? due to `Status` and `EventStatus`

**Comment by marandaneto** [comment]
@AbhiPrasad or @kamilogorek Since you both are JS magicians, would you be so kind to review this? thanks.

File: src/js/wrapper.ts (hunk 1 above)
Comment (kamilogorek) [reply src/js/wrapper.ts:87]: Yup - https://github.com/getsentry/sentry-javascript/blob/00d359d2defdb730829b700d5c324a2f9cef553d/packages/core/src/request.ts#L88

**Review by kamilogorek — APPROVED** [review]

**Comment by AbhiPrasad** [comment]
> minor bump, maybe even major

@marandaneto

Minor should be fine. The enum changes are simply deprecations, so everything is backwards compatible. They will only be formally removed in the major bump.

**Pull request merged**